### ISSUE TITLE
[feature] conan list is not ergonomic

### ISSUE THREAD

**Issue by pierricgimmig** [conan-io/conan#17299]
### Make `conan list` output all packages without requiring the "*"

Intuitively, I want to be able to list my conan packages using `conan list`

```
C:\Users\pierr\git
λ conan list
ERROR: Missing pattern or graph json file
```

I then have to go through the docs to find an example that tells me that I'm missing "*".

Why require the "*" argument? It's a small thing, but it's frustrating to the user, especially since Conan is close to Python, and we'd expect `conan list` to work like `pip list` does.

Suggestion, make `conan list` behave like `conan list *`

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @pierricgimmig 

Thanks for your suggestion.

I am proposing the change in https://github.com/conan-io/conan/pull/17300, but can't guarantee 100% it will move forward, I will check with the team, there might be other UX reasons why this wasn't already the default, there were even tests that were checking this behavior.

**Comment by pierricgimmig** [comment]
Thank you @memsharded !

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/17300 was merged, so this will be in next Conan 2.10.

Thanks for the feedback!

**Comment by pierricgimmig** [comment]
Thanks for the quick turnaround!
